Tmain/omit-long-patterns-etags.d/stdout-expected.txt

```diff
@@ -1,5 +1,5 @@
 
-input.sh,349
-                                                                                                func96(func961,0
+input.sh,351
+                                                                                                func96()func961,0
                                                                                                func95()func955,110
-                                                                                                 func97func979,219
+                                                                                                 func97(func979,219
```

Comment:
@masatake do we wanna change this test case to balance the previous off-by-one-ness?  I'm not sure, because now it properly checks the behavior when the line is the exactly as long, short and longer than the truncation, which seems a good thing to do, but you might remember if you had another goal here maybe.

Reply:
It seems that there a bug has been existed since Exuberant-crags about newline handling.
`line` returned from `readLineFromBypassAnyway` can end with or without a newline char.
Both can occurs. Though the code trims the line with `line[len - 1] = '\0''`.

To combine your fix, I think the diff should be:
```
diff --git a/main/writer-etags.c b/main/writer-etags.c
index 762c8378..dbd6470f 100644
--- a/main/writer-etags.c
+++ b/main/writer-etags.c
@@ -98,18 +98,18 @@ static int writeEtagsEntry (tagWriter *writer,
                long seekValue;
                char *const line =
                                readLineFromBypassAnyway (etags->vLine, tag, &seekValue);
-               if (line == NULL)
+               if (line == NULL || line[0] == '\0')
                        return 0;
 
                len = strlen (line);
 
                if (tag->truncateLineAfterTag)
                        truncateTagLineAfterTag (line, tag->name, true);
-               else
-                       line [len - 1] = '\0';
+               else if (line [len - 1] == '\n')
+                       line [--len] = '\0';
 
                if (Option.patternLengthLimit < len)
-                       line [Option.patternLengthLimit - 1] = '\0';
+                       line [Option.patternLengthLimit] = '\0';
 
                length = mio_printf (mio, "%s\177%s\001%lu,%ld\n", line,
                                tag->name, tag->lineNumber, seekValue);
```